tsdb/engine.go

```diff
@@ -1,892 +1,127 @@
-	SelectColumnCountWithOneValue = 2
-
-	// IgnoredChunkSize is what gets passed into Mapper.Begin for aggregate queries as they don't chunk points out
-	IgnoredChunkSize = 0
+var (
+	// ErrFormatNotFound is returned when no format can be determined from a path.
+	ErrFormatNotFound = errors.New("format not found")
 )
 
-// Mapper is the interface all Mapper types must implement.
-type Mapper interface {
-	Open() error
-	TagSets() []string
-	NextChunk() (interface{}, error)
-	Close()
-}
-
-// StatefulMapper encapsulates a Mapper and some state that the executor needs to
-// track for that mapper.
-type StatefulMapper struct {
-	Mapper
-	bufferedChunk *MapperOutput // Last read chunk.
-	drained       bool
-}
-
-// NextChunk wraps a RawMapper and some state.
-func (sm *StatefulMapper) NextChunk() (*MapperOutput, error) {
-	c, err := sm.Mapper.NextChunk()
-	if err != nil {
-		return nil, err
-	}
-	chunk, ok := c.(*MapperOutput)
-	if !ok {
-		if chunk == interface{}(nil) {
-			return nil, nil
-		}
-	}
-	return chunk, nil
-}
-
-type Executor struct {
-	stmt           *influxql.SelectStatement
-	mappers        []*StatefulMapper
-	chunkSize      int
-	limitedTagSets map[string]struct{} // Set tagsets for which data has reached the LIMIT.
-}
+// DefaultEngine is the default engine used by the shard when initializing.
+const DefaultEngine = "v1"
 
-// NewRawExecutor returns a new RawExecutor.
-func NewExecutor(stmt *influxql.SelectStatement, mappers []Mapper, chunkSize int) *Executor {
-	a := []*StatefulMapper{}
-	for _, m := range mappers {
-		a = append(a, &StatefulMapper{m, nil, false})
-	}
-	return &Executor{
-		stmt:           stmt,
-		mappers:        a,
-		chunkSize:      chunkSize,
-		limitedTagSets: make(map[string]struct{}),
-	}
-}
+// Engine represents a swappable storage engine for the shard.
+type Engine interface {
+	Open() error
+	Close() error
 
-// Execute begins execution of the query and returns a channel to receive rows.
-func (e *Executor) Execute() <-chan *influxql.Row {
-	// Create output channel and stream data in a separate goroutine.
-	out := make(chan *influxql.Row, 0)
+	SetLogOutput(io.Writer)
+	LoadMetadataIndex(index *DatabaseIndex, measurementFields map[string]*MeasurementFields) error
 
-	if (e.stmt.IsRawQuery && !e.stmt.HasDistinct()) || e.stmt.IsSimpleDerivative() {
-		go e.executeRaw(out)
-	} else {
-		go e.executeAggregate(out)
-	}
-	return out
-}
+	Begin(writable bool) (Tx, error)
+	DB() *bolt.DB
+	WritePoints(points []Point, measurementFieldsToSave map[string]*MeasurementFields, seriesToCreate []*SeriesCreate) error
+	DeleteSeries(keys []string) error
+	DeleteMeasurement(name string, seriesKeys []string) error
+	SeriesCount() (n int, err error)
 
-// mappersDrained returns whether all the executors Mappers have been drained of data.
-func (e *Executor) mappersDrained() bool {
-	for _, m := range e.mappers {
-		if !m.drained {
-			return false
-		}
-	}
-	return true
+	Flush(partitionFlushDelay time.Duration) error
```

Comment:
This was only being used by a test so I was able to remove it from the interface: 0c61e20.